### PR TITLE
Changed names in Grammar and added number types

### PR DESCRIPTION
**Pull request by lukasberglund**
Changed naming of Expressions and Types in Grammar; created seperate types for Int and Double in Grammar. 

The motivation behind creating Double and Float types is that in the type theory book, evaluators worked by converting more complex expressions into simpler expressions step by step until they became values. That means we have to have values be expressions, so a generic number type is not enough.

### CODE REVIEW

File: src/Grammar.hs [hunk 1]
```diff
@@ -30,67 +30,68 @@ instance Read Bits where
...
-  | Str
-  | Bool
-  | Void
-  | Array Type
+  | TDouble
+  | TChar
+  | TStr
+  | TBool
+  | TVoid
+  | TArray Type
```
Comment (ayazhafiz) [inline src/Grammar.hs:42]: :+1:

File: src/Grammar.hs [hunk 2]
```diff
@@ -30,67 +30,68 @@ instance Read Bits where
...
-  | Char Char
-  | Reference String
-  | Declaration
+  = EInt Int
+  | EDouble Float
+  | EString String
+  | EArray [Expression]
+  | EChar Char
+  | EReference String
+  | EDeclaration
```
Comment (ayazhafiz) [inline src/Grammar.hs:78]: I would prefer something like `IntExp`. What do you think?

Comment (lukasberglund) [reply src/Grammar.hs:78]: How about `IntExpr` since `Exp` could be `Expectation`?

**Comment by ayazhafiz** [comment]
sounds good

**Comment by lukasberglund** [comment]
@ayazhafiz what do you think of the naming of types by adding the `T` prefix?

**Comment by ayazhafiz** [comment]
I like it.

**Comment by ayazhafiz** [comment]
Rebasing and merging 🚀

**Pull request merged**